code/math/implementation.lisp

```diff
@@ -1,17 +1,173 @@
 (in-package #:quaviver/math)
 
+(deftype word (arithmetic-size &optional (count 1))
+   #+quaviver/bignum-elision
+   (ecase arithmetic-size
+     (32 (case count
+           (1 `(unsigned-byte 32))
+           (2 `(unsigned-byte 64))
+           (otherwise `(simple-array (unsigned-byte 64) (,(ceiling count 2))))))
+     (64 (if (eql count 1)
+             `(unsigned-byte 64)
+             `(simple-array (unsigned-byte 64) (,count))))
+     (128 `(simple-array (unsigned-byte 64) (,(ash count 1)))))
```

Comment:
This should probably be ``(128 `(unsigned-byte ,(* arithmetic-size count))))``. We are currently only need arithmetic size 128 on ECL, but having the 128-bit routines not have a no-bignum version will greatly simplify things, since we would need to have a custom LDB, ASH, etc.